### PR TITLE
Add support for Cisco's AsyncOS

### PR DESCRIPTION
**Pull request by network-guy**
This has been tested on the Web Security Appliance, version 10+. It should work with the E-mail Security Appliances and should be backwards compatible at least to version 9.0. 

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by laf** [comment]
This needs unit tests adding as well: http://docs.librenms.org/Developing/os/Test-Units/

**Comment by network-guy** [comment]
Should be added now. Had some issues with Git, but I think I got them worked out.

File: includes/polling/os/asyncos.inc.php [hunk 1]
```diff
@@ -20,3 +20,7 @@
 
 preg_match('/[[\w]+-[\w]+/', $serial, $regexp_results);
 $serial = $regexp_results[0];
+<<<<<<< HEAD
```
Comment (laf) [inline includes/polling/os/asyncos.inc.php]: This needs removing :)

File: tests/snmpsim/asyncos.snmprec [hunk 2]
```diff
@@ -0,0 +1,2 @@
+1.3.6.1.2.1.1.1.0|4|Cisco Model S390, AsyncOS Version: 10.1.1-230, Build Date: 2017-03-15, Serial #: 006BF1CD6AE0-FCH2044V0YB
```
Comment (laf) [inline tests/snmpsim/asyncos.snmprec]: If this is a valid serial # considering changing it.

**Comment by laf** [comment]
Thanks, just my comment on the serial number now.

**Comment by network-guy** [comment]
Changed, thanks.

**Pull request merged**